### PR TITLE
Add "Retry" button to regular JS error notices in the plugin - followup

### PR DESCRIPTION
**Pull request by techanvil**
## Summary

This is a followup for #5494 to address the issues raised here: https://github.com/google/site-kit-wp/issues/5494#issuecomment-1285721043

Addresses issue:

- #5494 

## Relevant technical choices

Added a `clearMatchingLegacyError` param to `clearError` in order to address a logical error within `createFetchStore`. See [here](https://github.com/google/site-kit-wp/pull/6047/files#r1003443085https://github.com/google/site-kit-wp/pull/6047/files#r1003443085) for more details.

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [ ] I have added a QA Brief on the issue linked above.
- [ ] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/googlesitekit/data/create-fetch-store.js [hunk 1]
```diff
@@ -157,7 +157,8 @@ export const createFetchStore = ( {
 			type: START_FETCH,
 		};
 
-		yield clearError( baseName, args );
+		// @TODO: remove clearMatchingLegacyError usage once all instances of the legacy behavior have been removed.
+		yield clearError( baseName, args, { clearMatchingLegacyError: true } );
```
Comment (techanvil) [inline assets/js/googlesitekit/data/create-fetch-store.js:161]: The `clearMatchingLegacyError` param is used here in order to address a logical error, where further down this function, when an error is received it's set in the store twice (once in `state.errors` keyed by the error key, and also in the single `state.error` property). However in the prelude here only the keyed error in `state.errors` is removed. This means that a stale fetch error can remain in state following a refetch.

This became evident when making the change to `ErrorNotice`, when successfully retrying a selector that results in a fetch, the stale error was still displayed.

Comment (tofumatt) [reply assets/js/googlesitekit/data/create-fetch-store.js:161]: From my perspective this isn't too bad, is well-tested, and is clearly documented.

I think this is a fine workaround for a temporary state of things 👍🏻

Comment (techanvil) [reply assets/js/googlesitekit/data/create-fetch-store.js:161]: Thanks for the feedback @tofumatt!

**Review by tofumatt — APPROVED** [review]
Code looks good to me, and tested with it working. 👍🏻

**Pull request merged**